app/models/concerns/category_hashtag.rb

```diff
@@ -7,19 +7,24 @@ module CategoryHashtag
 
   class_methods do
     def query_from_hashtag_slug(category_slug)
-      parent_slug, child_slug = category_slug.split(SEPARATOR, 2)
+      slug_path = category_slug.split(SEPARATOR)
 
-      categories = Category.where(slug: parent_slug)
+      return nil if slug_path.empty? || slug_path.size > 2
 
-      if child_slug
-        Category.where(slug: child_slug, parent_category_id: categories.select(:id)).first
-      else
-        categories.where(parent_category_id: nil).first
+      if SiteSetting.slug_generation_method == "encoded"
+        slug_path.map! { |slug| CGI.escape(slug) }
       end
-    end
-  end
 
-  def hashtag_slug
-    full_slug.split("-").last(2).join(SEPARATOR)
+      last_category_id = nil
+
+      slug_path.each do |slug|
+        query = Category.select(:id).where(slug: slug)
+        query = query.where(parent_category_id: last_category_id) if last_category_id.present?
+        return nil if query.blank?
```

Comment:
When would `query` be `blank?`

Reply:
Sure, that was needed for three level mentions.